pam/pam_test/module-transaction-dummy_test.go

```diff
@@ -0,0 +1,854 @@
+package pam_test
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/msteinert/pam"
+	"github.com/stretchr/testify/require"
+)
+
+func ptrValue[T any](value T) *T {
+	return &value
+}
+
+func bytesPointerDecoder(ptr pam.BinaryPointer) ([]byte, error) {
+	if ptr == nil {
+		return nil, nil
+	}
+	return *(*[]byte)(ptr), nil
+}
+
+func TestSetGetItem(t *testing.T) {
+	t.Parallel()
+	t.Cleanup(MaybeDoLeakCheck)
+
+	tests := map[string]struct {
+		item  pam.Item
+		value *string
+
+		wantValue    *string
+		wantGetError error
+		wantSetError error
+	}{
+		"Set user": {
+			item:  pam.User,
+			value: ptrValue("an user"),
+		},
+
+		"Returns empty when getting an unset user": {
+			item:      pam.User,
+			wantValue: ptrValue(""),
+		},
+
+		"Setting and getting an user": {
+			item:      pam.User,
+			value:     ptrValue("the-user"),
+			wantValue: ptrValue("the-user"),
+		},
+
+		// Error cases
+		"Error when setting invalid item": {
+			item:         pam.Item(-1),
+			value:        ptrValue("some value"),
+			wantSetError: pam.ErrBadItem,
+		},
+
+		"Error when getting invalid item": {
+			item:         pam.Item(-1),
+			wantGetError: pam.ErrBadItem,
+			wantValue:    ptrValue(""),
+		},
+	}
+
+	for name, tc := range tests {
+		tc := tc
+		t.Run(name, func(t *testing.T) {
+			t.Parallel()
+			t.Cleanup(MaybeDoLeakCheck)
+
+			// These test inputs can not be all unset
+			if tc.value == nil && tc.wantValue == nil {
```

Comment:
Let’s make the test as simple as possible. This is a programmer error, in the tests, let’s not complexify the syntax, I’m sure the developer will see the test failing and can fix it by him/herself.